openfisca_core/base_functions.py

```diff
@@ -51,7 +49,7 @@ def requested_period_last_value(formula, simulation, period, *extra_params, **kw
     known_periods = holder.get_known_periods()
     if not known_periods:
         return holder.default_array()
-    known_periods = sorted(known_periods, cmp = periods.compare_period_start, reverse = True)
+    known_periods = sorted(known_periods, key=lambda period: period.start, reverse = True)
```

Comment:
Calling the attribute directly seems to be the pythonic way to do it
https://wiki.python.org/moin/HowTo/Sorting#Sortingbykeys
```
>>> sorted(student_objects, key=lambda student: student.age)   # sort by age
[('dave', 'B', 10), ('jane', 'B', 12), ('john', 'A', 15)]
```
it there an added benefit to 
`periods = sorted(known_periods, key=lambda period: getattr(period, 'start'), reverse = True)`
that I'm not seeing ?

Reply:
I think @sandcha meant `known_periods = sorted(known_periods, key=operator.attrgetter('start'), reverse = True)`. See [doc](https://docs.python.org/3/library/operator.html#operator.attrgetter).

I find lambdas clearer, because more similar to other languages, but it seems to be [debated](https://utcc.utoronto.ca/~cks/space/blog/python/AttrgetterVsLamba)

Reply:
I'm keeping the lambda for now, however, let's test the performance of `operator.attrgetter('start')`vs object.start to see if we could make Openfisca more efficient .